refactor: standardize registry links

jniles opened the pull request:
This refactor updates the registries to have links to their receipts inline, as in the patient card.  Every registry has been updated in an effort to standardize their user experience.

Additionally, the patient links have been removed in favor of the drop down menu.

Finally, the other registries filter views are updated so that a full UUID is never shown to the client.

----

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

Commit:
feat(vouchers): link receipt w/ id column

Most registries will now link their receipts from the id column of the
registry.  This commit updates the voucher registry to do exactly that.

Commit:
feat(registries): link receipts from references

This commit implements the linked receipts from the grid reference
itself on both cash and invoices. This should save users a click.  Now
cash, patients, invoices, and vouchers are all uniform in their receipt
rendering.

It also makes sure that UUIDs are never rendered in their entirety to
the user.

Finally, the patient links are removed since they are available in the
"actions" dropdown.

sfount commented on client/src/js/services/CashService.js:
```diff
@@ -142,15 +142,21 @@ function CashService(Modal, Api, Exchange, Session, moment, $translate) {
 
     // returns columns from filters
     return columns.filter(function (column) {
+      var LIMIT_UUID_LENGTH = 6;
```
:+1: This is a good change to make for scenarios where we need to display a UUID. In the upcoming refactor on filters we should be able to display the patient's name to get around this bad UX.

sfount reviewed (commented):
Changes LGTM. One of our upcoming development focusses should be to make sure that all registries behave in a predictable way. 

As this pull request links documents through the reference link template, should this remove the double click to view document behaviour?

Commit:
refactor(registries): remove dbl-click handlers

This commit removes the double click handlers from the registries to ensure users are not
consistently opening the PDF documents instead of click drodpowns.  This functionality is surfaced
through the links on the voucher documents anyhow.

Closes #1232.
